frontend/src/features/map/layers/RegulatoryPreviewLayer.tsx

```diff
@@ -54,25 +56,31 @@ function UnmemoizedRegulatoryPreviewLayer({ map }: RegulatoryPreviewLayerProps)
 
     getVectorSource().clear()
 
-    if (regulatoryGeometriesToPreview && regulatoryGeometriesToPreview.length) {
-      const features = regulatoryGeometriesToPreview
-        .map(geometry => {
-          if (geometry) {
-            return new GeoJSON({
-              featureProjection: OPENLAYERS_PROJECTION
-            }).readFeature(geometry)
-          }
-
-          return null
-        })
-        .filter((feature): feature is Feature<Geometry> => Boolean(feature))
-
-      if (features?.length) {
-        getVectorSource().addFeatures(features)
-        dispatch(zoomInLayer({ feature: features[0] }))
-      }
+    if (!regulatoryZonesToPreview?.length) {
+      return
     }
-  }, [dispatch, map, regulatoryGeometriesToPreview])
+
+    const features = regulatoryZonesToPreview
```

Comment:
A good practice in components to reduce their internal logic is to export the transformations into a related component file like a `utils.ts`, which also makes these transformations easily testable at the same time.